zarr/storage.py

```diff
@@ -97,17 +217,255 @@ def init_store(store, shape, chunks, dtype=None, compression='default',
     )
     order = normalize_order(order)
 
-    # delete any pre-existing items in store
-    store.clear()
-
-    # initialise metadata
+    # initialize metadata
     meta = dict(shape=shape, chunks=chunks, dtype=dtype,
                 compression=compression, compression_opts=compression_opts,
                 fill_value=fill_value, order=order)
-    store['meta'] = encode_metadata(meta)
+    key = _path_to_prefix(path) + array_meta_key
+    store[key] = encode_array_metadata(meta)
+
+    # initialize attributes
+    key = _path_to_prefix(path) + attrs_key
+    store[key] = json.dumps(dict()).encode('ascii')
+
+
+# backwards compatibility
+init_store = init_array
+
+
+def init_group(store, overwrite=False, path=None):
+    """initialize a group store.
+
+    Parameters
+    ----------
+    store : MutableMapping
+        A mapping that supports string keys and byte sequence values.
+    overwrite : bool, optional
+        If True, erase all data in `store` prior to initialisation.
+    path : string, optional
+        Path under which array is stored.
+
+    """
+
+    # normalize path
+    path = normalize_storage_path(path)
+    
+    # guard conditions
+    if overwrite:
+        # attempt to delete any pre-existing items in store
+        rmdir(store, path)
+    elif contains_array(store, path):
+        raise ValueError('store contains an array')
+    elif contains_group(store, path):
+        raise ValueError('store contains a group')
+
+    # initialize metadata
+    # N.B., currently no metadata properties are needed, however there may
+    # be in future
+    meta = dict()
+    key = _path_to_prefix(path) + group_meta_key
+    store[key] = encode_group_metadata(meta)
+
+    # initialize attributes
+    key = _path_to_prefix(path) + attrs_key
+    store[key] = json.dumps(dict()).encode('ascii')
+
+
+def ensure_bytes(s):
+    if isinstance(s, binary_type):
+        return s
+    if hasattr(s, 'tobytes'):
+        return s.tobytes()
+    if PY2 and hasattr(s, 'tostring'):  # pragma: no cover
+        return s.tostring()
+    return memoryview(s).tobytes()
+
+
+def _dict_store_keys(d, prefix='', cls=dict):
+    for k in d.keys():
+        v = d[k]
+        if isinstance(v, cls):
+            for sk in _dict_store_keys(v, prefix + k + '/', cls):
+                yield sk
+        else:
+            yield prefix + k
+
+
+def buffersize(v):
+    from array import array as _stdlib_array
+    if PY2 and isinstance(v, _stdlib_array):  # pragma: no cover
+        # special case array.array because does not support buffer
+        # interface in PY2
+        return v.buffer_info()[1] * v.itemsize
+    else:
+        v = memoryview(v)
+        return reduce(operator.mul, v.shape) * v.itemsize
```

Comment:
What I try to store a scalar array, with shape `()`? Maybe it's worth more broadly checking that that works.